### PR TITLE
ipython 8.25.0

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-4933](https://anaconda.atlassian.net/browse/PKG-4933)
- [Upstream repository](https://github.com/ipython/ipython/tree/8.25.0)
- [Upstream changelog/diff](https://ipython.readthedocs.io/en/8.25.0/whatsnew/version8.html)
- Relevant dependency PRs:
  - https://github.com/AnacondaRecipes/traitlets-feedstock/pull/6

### Explanation of changes:

- Set version and sha256
- Remove outdated test parameters
- Dependency cleanup in test and run sections
- Sync `run_test.py` with conda forge version

[PKG-4933]: https://anaconda.atlassian.net/browse/PKG-4933?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ViridianMelody — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -55,11 +53,8 @@ test:
     - nbformat
     - numpy >=1.23
     - pandas
-    - pygments
-    - pytest-cov
     - trio
     {% endif %}
-
   commands:
```
Comment (danpetry) [inline recipe/meta.yaml:53]: does `MIGRATING=false run_test.py` work (and removing the migrating stuff elsewhere)? Maybe with a note about what the migrating thing means and syncing with conda-forge?

Comment (danpetry) [reply recipe/meta.yaml:53]: I found the whole migrating stuff overcomplicated

**Review by danpetry — APPROVED** [review]
good with a suggestion

File: recipe/run_test.py [hunk 2]
```diff
@@ -46,11 +46,6 @@
...
-
-    if MIGRATING:
-        print("This is a migration, skipping test suite! Put it back later!", flush=True)
-        sys.exit(0)
-    else:
-        print("Running pytest with args")
-        print(PYTEST_ARGS, flush=True)
-        sys.exit(subprocess.call(PYTEST_ARGS, cwd=str(Path(IPython.__file__).parent)))
\ No newline at end of file
+    print("Running pytest with args")
```
Comment (danpetry) [inline recipe/run_test.py:49]: It'd be less error-prone to update run_test.py next time if you kept it the same as conda-forge

Comment (danpetry) [reply recipe/run_test.py:49]: you could do `MIGRATING=false run_test.py`, or `export MIGRATING=false` in the test commands

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml:53]: If I recall it's for their python buildout. ipython is a somewhat central dependency which you need built before the test dependencies it uses.

Comment (cbouss) [reply recipe/meta.yaml:53]: The way our buildout was done, this mechanism was not required.

File: recipe/run_test.py (hunk 2 above)
Comment (SC426) [reply recipe/run_test.py:49]: `MIGRATING=false run_test.py` does not work

Comment (SC426) [reply recipe/run_test.py:49]: I opted to remove everything since it still builds

Comment (SC426) [reply recipe/run_test.py:49]: Added your second suggestion but am unsure why it won't work for windows

**Pull request merged**